admin/products_price_manager.php

```diff
@@ -106,54 +103,37 @@
-                  products_mixed_discount_quantity = :discQty:
+                  products_status = " . (int)$_POST['products_status'] . ",
+                  products_quantity_order_min = " . (float)$_POST['products_quantity_order_min'] . ",
+                  products_quantity_order_units = " . (float)$_POST['products_quantity_order_units'] . ",
+                  products_quantity_order_max = " . (float)$_POST['products_quantity_order_max'] . ",
+                  product_is_free = " . (int)$_POST['product_is_free'] . ",
+                  product_is_call = " . (int)$_POST['product_is_call'] . ",
+                  products_quantity_mixed = " . (int)$_POST['products_quantity_mixed'] . ",
+                  products_priced_by_attribute = " . (int)$_POST['products_priced_by_attribute'] . ",
+                  products_discount_type = " . (int)$_POST['products_discount_type'] . ",
+                  products_discount_type_from = " . (int)$_POST['products_discount_type_from'] . ",
+                  products_price_sorter = " . (int)$products_price_sorter . ",
+                  master_categories_id = " . (int)$master_categories_id . ",
+                  products_mixed_discount_quantity = " . (int)$_POST['products_mixed_discount_quantity'] . "
               WHERE products_id = " . (int)$products_filter;
 
-      $sql = $db->bindVars($sql, ':price:', $_POST['products_price'], 'string');
-      $sql = $db->bindVars($sql, ':taxClass:', $_POST['products_tax_class_id'], 'integer');
-      $sql = $db->bindVars($sql, ':dateAvailable:', $products_date_available, 'string');
-      $sql = $db->bindVars($sql, ':status:', $_POST['products_status'], 'integer');
-      $sql = $db->bindVars($sql, ':orderMin:', $_POST['products_quantity_order_min'], 'string');
-      $sql = $db->bindVars($sql, ':orderUnits:', $_POST['products_quantity_order_units'], 'string');
-      $sql = $db->bindVars($sql, ':orderMax:', $_POST['products_quantity_order_max'], 'string');
-      $sql = $db->bindVars($sql, ':isFree:', $_POST['product_is_free'], 'integer');
-      $sql = $db->bindVars($sql, ':isCall:', $_POST['product_is_call'], 'integer');
-      $sql = $db->bindVars($sql, ':qtyMixed:', $_POST['products_quantity_mixed'], 'integer');
-      $sql = $db->bindVars($sql, ':pricedByAttr:', $_POST['products_priced_by_attribute'], 'integer');
-      $sql = $db->bindVars($sql, ':discType:', (isset($_POST['products_discount_type']) ? $_POST['products_discount_type'] : 0), 'integer');
-      $sql = $db->bindVars($sql, ':discTypeFrom:', (isset($_POST['products_discount_type_from']) ? $_POST['products_discount_type_from'] : 0), 'integer');
-      $sql = $db->bindVars($sql, ':discPriceSorter:', $products_price_sorter, 'string');
-      $sql = $db->bindVars($sql, ':masterCatId:', $master_categories_id, 'integer');
-      $sql = $db->bindVars($sql, ':discQty:', (isset($_POST['products_mixed_discount_quantity']) ? $_POST['products_mixed_discount_quantity'] : 0), 'integer');
```

Comment:
I'm not sure I understand why you removed the `bindVars` implementation here.

What problem did it create that made you decide to remove it completely?

Reply:
I removed it because all are simple integers or floats, which can be casted directly. No need to use ```$db->bindVars``` which does the same with a detour. Also the strings are floats

Reply:
Thank you for your efforts. I wish you hadn't done that. I will probably change it back because it is cleaner to read a mass group of bindVars calls than it is to read inline casts interpolated with query strings. I'll have a chat with @zcwilt first.